include/vapor/ArbitrarilyOrientedRegularGrid.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <ostream>
+#include <vector>
+#include <vapor/common.h>
+#include <vapor/RegularGrid.h>
+#include <glm/glm.hpp>
+
+// clang-format off
+
+struct planeDescription {
+    size_t sideSize;
+    std::vector<double> origin;
+    std::vector<double> rotation;
+    VAPoR::CoordType boxMin;
+    VAPoR::CoordType boxMax;
+    VAPoR::CoordType domainMin;
+    VAPoR::CoordType domainMax;
+};
+
+namespace VAPoR {
+
+//! Create a 2D RegularGrid that is rotated along an arbitrary orientation.  This new grid
+//! samples a given 3D Grid object along a plane, through the entire domain.  The "planeDescription" 
+//! variable includes parameters that define the 2D plane's origin, rotation, and valid extents.  Points on the
+//! 2D RegularGrid that are outside of the valid extents will be assigned as
+//! missingValues.
+//!
+//! Note: The boxMin and boxMax values within the planeDescription should come from
+//! the current RenderParams' Box class.  The domainMin and domainMax values should
+//! come from the DataMgr::GetVariableExtents function.  Taking these values from
+//! the given 3D grid can produce erroneous results when the 3D grid's region is reduced.
+//!
+//! \param[in] grid3d A variable's 3D grid to be sampled along a plane, to generate a 2D grid object
+//! \param[in] description A set of std::vector<dobule> values that define the origin, rotation, and extents of the returned 2D grid
+//! \param[in] dims The sampling rate that the new grid will be defined upon, along its X and Y axes
+//! \param[out] data An array of sideSize*sideSize values that contain floating point values of the sampled data.
+//!
+
+class VDF_API ArbitrarilyOrientedRegularGrid : public RegularGrid {
```

Comment:
The Grid and RegularGrid class define several methods that operate on CoordType arguments. Other than GetUserCoordinates, which you have redefined, the others will produce incorrect results. We said that we would not bother to reimplement these until they are needed. But you should at lease have them throw an assert or something in case someone does try to use them.

Reply:
My original intent was to delete all unused functions but I ran into compiler errors when deleting virtual functions.  Looking at it again, I'm seeing that I can delete functions as long as I remove the const qualifier on them.  I'm not sure why.  A deleted function can't even be called, no less modify an object.  Who knows...

I've gone ahead and deleted the other forms of GetUserCoordinates().

PS - There are dozens of other public functions that I'm not certain will function either.  I can test and delete these too but it looks like it might be a more substantial effort.

Reply:
I don't think you have to test them. You can assume that they're going to break if they are operating in 2D space.

Reply:
Updated by deleting functions that reference the CoordType datatype.